# Melbourne parkruns

Step one of my project is figuring what parkruns are in the Greater Melbourne area as defined by ~~the State Revenue Office: https://www.sro.vic.gov.au/greater-melbourne-map-and-urban-zones#current-greater-melbourne~~ OpenStreetMap: https://www.openstreetmap.org/relation/4246124#map=9/-37.952/145.319

Luckily all parkrun events worldwide are referenced in this json file: https://images.parkrun.com/events.json so that's the first dataset we'll be playing with.

In [1]:
# Was expecting I'd be scraping the website initally, but turns out there's a nice json file I can look at instead.

import geopandas as gpd

url = "https://images.parkrun.com/events.json"
gdf = gpd.read_file(url)

gdf.head()

eventname             EventLongName    EventShortName  \
0          bushy             Bushy parkrun        Bushy Park   
1      wimbledon  Wimbledon Common parkrun  Wimbledon Common   
2       banstead    Banstead Woods parkrun    Banstead Woods   
3       richmond          Richmond parkrun     Richmond Park   
4  woodhousemoor    Woodhouse Moor parkrun    Woodhouse Moor   

  LocalisedEventLongName  countrycode  seriesid  \
0                   None           97         1   
1                   None           97         1   
2                   None           97         1   
3                   None           97         1   
4                   None           97         1   

                         EventLocation                   geometry  
0               Bushy Park, Teddington  POINT (-0.33579 51.41099)  
1                     Wimbledon Common  POINT (-0.23222 51.44208)  
2             Banstead Woods, Coulsdon  POINT (-0.18422 51.30765)  
3  Richmond Park, Richmond upon Thames  POINT (-0.29289 51.45196)  
4                Woodhouse Moor, Leeds  POINT (-1.56006 53.80858)

In [2]:
gdf.shape

(2727, 8)

In [3]:
gdf.columns

Index(['eventname', 'EventLongName', 'EventShortName',
       'LocalisedEventLongName', 'countrycode', 'seriesid', 'EventLocation',
       'geometry'],
      dtype='object')

In [4]:
gdf.describe

<bound method NDFrame.describe of                   eventname                     EventLongName  \
0                     bushy                     Bushy parkrun   
1                 wimbledon          Wimbledon Common parkrun   
2                  banstead            Banstead Woods parkrun   
3                  richmond                  Richmond parkrun   
4             woodhousemoor            Woodhouse Moor parkrun   
...                     ...                               ...   
2722  lakemaryanntingkkarli  Lake Mary Ann Tingkkarli parkrun   
2723       christchurchpark         Christchurch Park parkrun   
2724    saigawaryokuchikoen     Saigawa ryokuchi koen parkrun   
2725   mamquamspawningtrail    Mamquam Spawning Trail parkrun   
2726               comopark                 Como Park parkrun   

                EventShortName LocalisedEventLongName  countrycode  seriesid  \
0                   Bushy Park                   None           97         1   
1             Wimbledon C

**This gives me all the parkrun events worldwide but I'm only interested in parkruns in Melbourne so first I'm going to filter the dataset to only give me Australian parkruns.**

In [5]:
# From examining the dataset I can see that Australia's country code is 3.

au_gdf = gdf[gdf['countrycode'] == 3]

au_gdf.head()

eventname              EventLongName    EventShortName  \
21          stpeters          St Peters parkrun         St Peters   
46   albertmelbourne  Albert parkrun, Melbourne  Albert Melbourne   
177          newfarm           New Farm parkrun          New Farm   
185        mainbeach         Main Beach parkrun        Main Beach   
304            kirra              Kirra parkrun             Kirra   

    LocalisedEventLongName  countrycode  seriesid  \
21                    None            3         1   
46                    None            3         1   
177                   None            3         1   
185                   None            3         1   
304                   None            3         1   

                              EventLocation                     geometry  
21           Sydney Park, St Peters, Sydney  POINT (151.18219 -33.90766)  
46        Albert Park, Melbourne, Australia  POINT (144.96312 -37.84352)  
177                      New Farm, Brisbane  POINT (153.05164 -27.47113)  
185                   Main Beach, Goldcoast  POINT (153.42888 -27.97271)  
304  Kirra Beach, Gold Coast QLD, Australia  POINT (153.52278 -28.16676)

In [6]:
# According to https://www.parkrun.com.au/ there are 518 locations in Australia, have I got all of them in my dataset?

f"Number of Australian parkruns: {len(au_gdf)}"

'Number of Australian parkruns: 520'

In [39]:
# Where is that extra (UPDATE: extra two) parkrun coming from? I will export my au_gdf to csv to manually check. 

au_gdf.to_csv("au-parkruns.csv", index=False)

In [8]:
# They all look like legit parkruns to me from a quick skim of the csv. Thought of something else. Are any of the parkruns in my dataset at the same location?

au_gdf['geometry'].nunique()

520

Nope to that last question so not sure where that extra rogue parkrun is coming from -- may need to ask parkrun Australia directly if their website is correct.

**UPDATE:** There is both a parkrun event and a junior event at Westerfolds Park, they have slightly different long/lats hence .nunique didn't pick them up but that could be why there's one more parkrun event to locations. And since I first ran the data the number of parkruns has gone from 519 to 520 which what looks like a new one near Tennant Creek, NT so the stats at the bottom of the website may have not be updated to reflect that.

In [11]:
# Have identified 5 junior parkruns as part of my dataset that I want to exclude. 

au_pk_gdf = au_gdf[~au_gdf['EventLongName'].str.contains('junior parkrun', na=False)]
f"Number of Australian (not junior) parkruns: {len(au_pk_gdf)}"

'Number of Australian (not junior) parkruns: 515'

## Now to filter the dataset further to get Melbourne parkruns only

I was thinking of getting a rough approximation of events in Melbourne by creating a bounding box using the lat/long coordinates of Greater Melbourne's northernmost, southernmost, easternmost and westernmost points and then checking manually to remove events outside the city bounds. 

However ChatGPT suggested another option: Use a precise shapefile or GeoJSON of Greater Melbourne and filter using that. I've never done this before but I'm going to give it a try.

I've found this Major urban area geojson on the Data Vic website: https://discover.data.vic.gov.au/dataset/major-urban-area-location-polygons-and-table/resource/eea2ce6a-68b0-49d2-827c-82feb335739c

In [13]:
vic_urban_boundaries = gpd.read_file("DataVic_MAJOR_URBAN_AREA.geojson")
vic_urban_boundaries

OBJECTID                    MUA_Name            LGA_Name  \
0          1                     Bendigo     Greater Bendigo   
1          2                     Bendigo     Greater Bendigo   
2          1                   Melbourne             Various   
3          2                   Melbourne             Various   
4          3                   Melbourne             Various   
..       ...                         ...                 ...   
67        18                     Morwell             Latrobe   
68        15                        Lara     Greater Geelong   
69        12  Drysdale - Clifton Springs     Greater Geelong   
70        30                 Maryborough  Central Goldfields   
71        16                     Leopold     Greater Geelong   

                            Source  \
0                      Bendigo UGB   
1                      Bendigo UGB   
2                    Melbourne UGB   
3                    Melbourne UGB   
4                    Melbourne UGB   
..                             ...   
67  Derived from UCL and PLAN_ZONE   
68  Derived from UCL and PLAN_ZONE   
69  Derived from UCL and PLAN_ZONE   
70  Derived from UCL and PLAN_ZONE   
71  Derived from UCL and PLAN_ZONE   

                                             geometry  
0   POLYGON ((144.18505 -36.73521, 144.1859 -36.73...  
1   POLYGON ((144.37767 -36.81894, 144.38078 -36.8...  
2   POLYGON ((144.70389 -38.3209, 144.70586 -38.32...  
3   POLYGON ((144.99114 -38.36436, 144.99111 -38.3...  
4   POLYGON ((145.02212 -38.38637, 145.02294 -38.3...  
..                                                ...  
67  POLYGON ((146.38503 -38.22981, 146.38502 -38.2...  
68  POLYGON ((144.3898 -38.03458, 144.38965 -38.03...  
69  POLYGON ((144.54465 -38.16444, 144.54464 -38.1...  
70  POLYGON ((143.74628 -37.06445, 143.74628 -37.0...  
71  POLYGON ((144.48314 -38.17795, 144.48314 -38.1...  

[72 rows x 5 columns]

In [14]:
vic_urban_boundaries['MUA_Name'].nunique()

34

In [15]:
vic_urban_boundaries.columns

Index(['OBJECTID', 'MUA_Name', 'LGA_Name', 'Source', 'geometry'], dtype='object')

In [16]:
# I'm filtering it to get only the urban areas of Melbourne.

melb_boundary = vic_urban_boundaries[vic_urban_boundaries['MUA_Name'] == 'Melbourne']
melb_boundary

OBJECTID   MUA_Name LGA_Name         Source  \
2          1  Melbourne  Various  Melbourne UGB   
3          2  Melbourne  Various  Melbourne UGB   
4          3  Melbourne  Various  Melbourne UGB   
5          4  Melbourne  Various  Melbourne UGB   
6          5  Melbourne  Various  Melbourne UGB   
7          6  Melbourne  Various  Melbourne UGB   
8          7  Melbourne  Various  Melbourne UGB   
9          8  Melbourne  Various  Melbourne UGB   
10         9  Melbourne  Various  Melbourne UGB   
11        10  Melbourne  Various  Melbourne UGB   
12        11  Melbourne  Various  Melbourne UGB   
13        12  Melbourne  Various  Melbourne UGB   
14        13  Melbourne  Various  Melbourne UGB   
15        14  Melbourne  Various  Melbourne UGB   
16        15  Melbourne  Various  Melbourne UGB   
17        16  Melbourne  Various  Melbourne UGB   
18        17  Melbourne  Various  Melbourne UGB   
19        18  Melbourne  Various  Melbourne UGB   
20        19  Melbourne  Various  Melbourne UGB   
21        20  Melbourne  Various  Melbourne UGB   
22        21  Melbourne  Various  Melbourne UGB   
23        22  Melbourne  Various  Melbourne UGB   
24        23  Melbourne  Various  Melbourne UGB   
25        24  Melbourne  Various  Melbourne UGB   
26        25  Melbourne  Various  Melbourne UGB   
27        26  Melbourne  Various  Melbourne UGB   
28        27  Melbourne  Various  Melbourne UGB   
29        28  Melbourne  Various  Melbourne UGB   
30        29  Melbourne  Various  Melbourne UGB   
31        30  Melbourne  Various  Melbourne UGB   
32        31  Melbourne  Various  Melbourne UGB   
33        32  Melbourne  Various  Melbourne UGB   
34        33  Melbourne  Various  Melbourne UGB   
35        34  Melbourne  Various  Melbourne UGB   
36        35  Melbourne  Various  Melbourne UGB   
37        36  Melbourne  Various  Melbourne UGB   
38        37  Melbourne  Various  Melbourne UGB   
39        38  Melbourne  Various  Melbourne UGB   

                                             geometry  
2   POLYGON ((144.70389 -38.3209, 144.70586 -38.32...  
3   POLYGON ((144.99114 -38.36436, 144.99111 -38.3...  
4   POLYGON ((145.02212 -38.38637, 145.02294 -38.3...  
5   POLYGON ((145.06583 -38.42529, 145.06423 -38.4...  
6   POLYGON ((145.00695 -38.48095, 145.00693 -38.4...  
7   POLYGON ((145.11631 -38.3635, 145.11268 -38.37...  
8   POLYGON ((145.14823 -38.3923, 145.14871 -38.39...  
9   POLYGON ((145.22083 -38.37542, 145.22173 -38.3...  
10  POLYGON ((145.16682 -38.27577, 145.16682 -38.2...  
11  POLYGON ((145.14254 -38.19629, 145.14255 -38.1...  
12  POLYGON ((145.47564 -38.19769, 145.47557 -38.1...  
13  POLYGON ((145.57212 -38.27903, 145.573 -38.274...  
14  POLYGON ((145.66825 -38.0799, 145.66818 -38.07...  
15  POLYGON ((145.70366 -38.08735, 145.7037 -38.09...  
16  POLYGON ((145.40519 -37.99997, 145.40825 -38.0...  
17  POLYGON ((145.55007 -37.94973, 145.54999 -37.9...  
18  POLYGON ((145.41988 -37.86335, 145.41865 -37.8...  
19  POLYGON ((145.41025 -37.81991, 145.41074 -37.8...  
20  POLYGON ((145.68475 -37.74908, 145.68469 -37.7...  
21  POLYGON ((145.60204 -37.78337, 145.60218 -37.7...  
22  POLYGON ((145.64569 -37.78001, 145.64524 -37.7...  
23  MULTIPOLYGON (((145.64609 -37.75253, 145.65477...  
24  POLYGON ((145.18345 -37.63998, 145.18687 -37.6...  
25  POLYGON ((145.23895 -37.64507, 145.23968 -37.6...  
26  POLYGON ((145.26408 -37.60675, 145.26595 -37.6...  
27  POLYGON ((145.50734 -37.66832, 145.5075 -37.66...  
28  POLYGON ((145.51092 -37.65171, 145.51092 -37.6...  
29  POLYGON ((145.37083 -37.64451, 145.37064 -37.6...  
30  POLYGON ((145.38824 -37.72436, 145.38798 -37.7...  
31  POLYGON ((145.42626 -37.77386, 145.42612 -37.7...  
32  POLYGON ((145.46647 -37.76777, 145.46562 -37.7...  
33  POLYGON ((145.50216 -37.77023, 145.502 -37.770...  
34  POLYGON ((145.54477 -37.77811, 145.5447 -37.77...  
35  POLYGON ((145.5575 -37.78203, 145.5579 -37.782...  
36  POLYGON ((144.70896 -37.62893, 144.71659 -37.6... 

In [17]:
melb_boundary.to_file("DataVic_greater_melb.geojson", driver="GeoJSON")

## It turns out the concept of Greater Melbourne was more complicated that I initially realised!

Ask OpenStreetMap for a map of Greater Melbourne and it returns this: https://www.openstreetmap.org/relation/4246124#map=9/-37.952/145.319  
(Which I turned into a GeoJSON file here: https://polygons.openstreetmap.fr/?id=4246124)

But the SRO's version is more bitsy (area in blue): https://www.sro.vic.gov.au/greater-melbourne-map-and-urban-zones#current-greater-melbourne

It appears that OpenStreetMap takes the blue and green areas of the map as Greater Melbourne whereas the SRO clearly defines the green areas as "Councils that have land both inside and outside the Urban Growth Boundary. The green denotes the parts of these councils falling outside the Urban Growth Boundary and therefore outside greater Melbourne."

As I'm using state government data I'm following the SRO's definition of Greater Melbourne for this analysis, while noting that such things are subjective and people may consider they live in Greater Melbourne even when by this definition they don't.

**UPDATE:** After manually checking the list of Melbourne parkruns the DataVic boundaries gave me against the parkrun website, it led to some parkruns being left out that a reasonable Melburnian would likely consider part of the Greater Melbourne area, hence I've switched to using OpenStreetMap's definition of Greater Melbourne and updating this analysis to reflect that. 

In [18]:
greater_melbourne_boundary = gpd.read_file("OSM_greater_melbourne.geojson")
greater_melbourne_boundary

geometry
0  MULTIPOLYGON (((144.44405 -37.86413, 144.44898...

In [19]:
greater_melbourne_boundary.columns

Index(['geometry'], dtype='object')

In [20]:
# Checking the CRS for each of my geopandas are the same.

greater_melbourne_boundary.crs

<Geographic 2D CRS: EPSG:4326>
Name: WGS 84
Axis Info [ellipsoidal]:
- Lat[north]: Geodetic latitude (degree)
- Lon[east]: Geodetic longitude (degree)
Area of Use:
- name: World.
- bounds: (-180.0, -90.0, 180.0, 90.0)
Datum: World Geodetic System 1984 ensemble
- Ellipsoid: WGS 84
- Prime Meridian: Greenwich

In [21]:
au_pk_gdf.crs

<Geographic 2D CRS: EPSG:4326>
Name: WGS 84
Axis Info [ellipsoidal]:
- Lat[north]: Geodetic latitude (degree)
- Lon[east]: Geodetic longitude (degree)
Area of Use:
- name: World.
- bounds: (-180.0, -90.0, 180.0, 90.0)
Datum: World Geodetic System 1984 ensemble
- Ellipsoid: WGS 84
- Prime Meridian: Greenwich

**Used my GeoJSON of Greater Melbourne to spatially join and filter parkruns within that region. Using .sjoin instead of .merge because I'm joining based on spatial relationships (ie. whether a parkrun is in Greater Melbourne or not) rather than a shared column.**

In [ ]:
# Rename geometry column in melb_boundary gdf so that it doesn't get lost when I join the two.

# melb_boundary['boundary_geom'] = melb_boundary.geometry
# melb_boundary

# Not bothering with this any more as it's only one polygon and two geometry columns can cause problems later. Can easily combine two GeoJSON files in geojson.io if I need to. 

In [22]:
melb_parkruns = gpd.sjoin(au_pk_gdf, greater_melbourne_boundary, predicate='within')
melb_parkruns

eventname  \
46                   albertmelbourne   
472                        highlands   
485                hastingsforeshore   
504                     diamondcreek   
520                   berwicksprings   
521                        pointcook   
536                      westerfolds   
572                    lillydalelake   
633                      maribyrnong   
651                         pakenham   
666                     toolerncreek   
682                       froghollow   
736                        parkville   
751                         brimbank   
766                           coburg   
804                            jells   
841                      altonabeach   
948                      wyndhamvale   
989                          studley   
1004                           lalor   
1062             chelseabicentennial   
1063                   wilsonbotanic   
1080                       karkarook   
1101                    mullummullum   
1114                         darebin   
1164                          mernda   
1191                    newportlakes   
1207                         sunbury   
1214                  marriottwaters   
1253                  gardinerscreek   
1282                         rosebud   
1550                  kmreedyreserve   
1615                birdslandreserve   
1745                   berwickwaters   
1788                          aurora   
1855              warringalparklands   
1857                       dandenong   
1877             cyrilcurtainreserve   
1977         cascadesonclydewetlands   
2008         dorsetrecreationreserve   
2076  frankstonnatureconservationres   
2105                lewisparkreserve   
2125          warrandyteriverreserve   
2148                 kirkdalereserve   
2319           woodlandshistoricpark   
2323                    maroondahdam   
2600                     emeraldlake   
2613                  aintreereserve   
2716                       belvedere   

                                      EventLongName  \
46                        Albert parkrun, Melbourne   
472                               Highlands parkrun   
485                      Hastings Foreshore parkrun   
504                           Diamond Creek parkrun   
520                         Berwick Springs parkrun   
521                              Point Cook parkrun   
536                             Westerfolds parkrun   
572                          Lillydale Lake parkrun   
633                             Maribyrnong parkrun   
651                                Pakenham parkrun   
666                           Toolern Creek parkrun   
682                             Frog Hollow parkrun   
736                               Parkville parkrun   
751                                Brimbank parkrun   
766                                  Coburg parkrun   
804                                   Jells parkrun   
841                            Altona Beach parkrun   
948                            Wyndham Vale parkrun   
989                                 Studley parkrun   
1004                                  Lalor parkrun   
1062                   Chelsea Bicentennial parkrun   
1063                         Wilson Botanic parkrun   
1080                              Karkarook parkrun   
1101                          Mullum Mullum parkrun   
1114                                Darebin parkrun   
1164                                 Mernda parkrun   
1191                          Newport Lakes parkrun   
1207                                Sunbury parkrun   
1214                        Marriott Waters parkrun   
1253                        Gardiners Creek parkrun   
1282                                Rosebud parkrun   
1550                       KM Reedy Reserve parkrun   
1615                      Birdsland Reserve parkrun   
1745                         Berwick Waters parkrun   
1788                                 Aurora parkrun   
1855                    Warringal Parklands pa

In [23]:
melb_parkruns.shape

(49, 9)

In [24]:
melb_parkruns = melb_parkruns.drop(columns=['index_right', 'LocalisedEventLongName'])
melb_parkruns

eventname  \
46                   albertmelbourne   
472                        highlands   
485                hastingsforeshore   
504                     diamondcreek   
520                   berwicksprings   
521                        pointcook   
536                      westerfolds   
572                    lillydalelake   
633                      maribyrnong   
651                         pakenham   
666                     toolerncreek   
682                       froghollow   
736                        parkville   
751                         brimbank   
766                           coburg   
804                            jells   
841                      altonabeach   
948                      wyndhamvale   
989                          studley   
1004                           lalor   
1062             chelseabicentennial   
1063                   wilsonbotanic   
1080                       karkarook   
1101                    mullummullum   
1114                         darebin   
1164                          mernda   
1191                    newportlakes   
1207                         sunbury   
1214                  marriottwaters   
1253                  gardinerscreek   
1282                         rosebud   
1550                  kmreedyreserve   
1615                birdslandreserve   
1745                   berwickwaters   
1788                          aurora   
1855              warringalparklands   
1857                       dandenong   
1877             cyrilcurtainreserve   
1977         cascadesonclydewetlands   
2008         dorsetrecreationreserve   
2076  frankstonnatureconservationres   
2105                lewisparkreserve   
2125          warrandyteriverreserve   
2148                 kirkdalereserve   
2319           woodlandshistoricpark   
2323                    maroondahdam   
2600                     emeraldlake   
2613                  aintreereserve   
2716                       belvedere   

                                      EventLongName  \
46                        Albert parkrun, Melbourne   
472                               Highlands parkrun   
485                      Hastings Foreshore parkrun   
504                           Diamond Creek parkrun   
520                         Berwick Springs parkrun   
521                              Point Cook parkrun   
536                             Westerfolds parkrun   
572                          Lillydale Lake parkrun   
633                             Maribyrnong parkrun   
651                                Pakenham parkrun   
666                           Toolern Creek parkrun   
682                             Frog Hollow parkrun   
736                               Parkville parkrun   
751                                Brimbank parkrun   
766                                  Coburg parkrun   
804                                   Jells parkrun   
841                            Altona Beach parkrun   
948                            Wyndham Vale parkrun   
989                                 Studley parkrun   
1004                                  Lalor parkrun   
1062                   Chelsea Bicentennial parkrun   
1063                         Wilson Botanic parkrun   
1080                              Karkarook parkrun   
1101                          Mullum Mullum parkrun   
1114                                Darebin parkrun   
1164                                 Mernda parkrun   
1191                          Newport Lakes parkrun   
1207                                Sunbury parkrun   
1214                        Marriott Waters parkrun   
1253                        Gardiners Creek parkrun   
1282                                Rosebud parkrun   
1550                       KM Reedy Reserve parkrun   
1615                      Birdsland Reserve parkrun   
1745                         Berwick Waters parkrun   
1788                                 Aurora parkrun   
1855                    Warringal Parklands pa

In [25]:
# Fix broken character in one of the cells, likely caused by the curly apostropher on their page. 

melb_parkruns['EventLocation'] = melb_parkruns['EventLocation'].str.replace("’", "'", regex=False)
melb_parkruns

eventname  \
46                   albertmelbourne   
472                        highlands   
485                hastingsforeshore   
504                     diamondcreek   
520                   berwicksprings   
521                        pointcook   
536                      westerfolds   
572                    lillydalelake   
633                      maribyrnong   
651                         pakenham   
666                     toolerncreek   
682                       froghollow   
736                        parkville   
751                         brimbank   
766                           coburg   
804                            jells   
841                      altonabeach   
948                      wyndhamvale   
989                          studley   
1004                           lalor   
1062             chelseabicentennial   
1063                   wilsonbotanic   
1080                       karkarook   
1101                    mullummullum   
1114                         darebin   
1164                          mernda   
1191                    newportlakes   
1207                         sunbury   
1214                  marriottwaters   
1253                  gardinerscreek   
1282                         rosebud   
1550                  kmreedyreserve   
1615                birdslandreserve   
1745                   berwickwaters   
1788                          aurora   
1855              warringalparklands   
1857                       dandenong   
1877             cyrilcurtainreserve   
1977         cascadesonclydewetlands   
2008         dorsetrecreationreserve   
2076  frankstonnatureconservationres   
2105                lewisparkreserve   
2125          warrandyteriverreserve   
2148                 kirkdalereserve   
2319           woodlandshistoricpark   
2323                    maroondahdam   
2600                     emeraldlake   
2613                  aintreereserve   
2716                       belvedere   

                                      EventLongName  \
46                        Albert parkrun, Melbourne   
472                               Highlands parkrun   
485                      Hastings Foreshore parkrun   
504                           Diamond Creek parkrun   
520                         Berwick Springs parkrun   
521                              Point Cook parkrun   
536                             Westerfolds parkrun   
572                          Lillydale Lake parkrun   
633                             Maribyrnong parkrun   
651                                Pakenham parkrun   
666                           Toolern Creek parkrun   
682                             Frog Hollow parkrun   
736                               Parkville parkrun   
751                                Brimbank parkrun   
766                                  Coburg parkrun   
804                                   Jells parkrun   
841                            Altona Beach parkrun   
948                            Wyndham Vale parkrun   
989                                 Studley parkrun   
1004                                  Lalor parkrun   
1062                   Chelsea Bicentennial parkrun   
1063                         Wilson Botanic parkrun   
1080                              Karkarook parkrun   
1101                          Mullum Mullum parkrun   
1114                                Darebin parkrun   
1164                                 Mernda parkrun   
1191                          Newport Lakes parkrun   
1207                                Sunbury parkrun   
1214                        Marriott Waters parkrun   
1253                        Gardiners Creek parkrun   
1282                                Rosebud parkrun   
1550                       KM Reedy Reserve parkrun   
1615                      Birdsland Reserve parkrun   
1745                         Berwick Waters parkrun   
1788                                 Aurora parkrun   
1855                    Warringal Parklands pa

In [26]:
melb_parkruns.dtypes

eventname           object
EventLongName       object
EventShortName      object
countrycode          int32
seriesid             int32
EventLocation       object
geometry          geometry
dtype: object

In [27]:
melb_parkruns.to_file("melb_parkruns.geojson", driver="GeoJSON")

In [28]:
melb_parkruns['EventLongName']

46                          Albert parkrun, Melbourne
472                                 Highlands parkrun
485                        Hastings Foreshore parkrun
504                             Diamond Creek parkrun
520                           Berwick Springs parkrun
521                                Point Cook parkrun
536                               Westerfolds parkrun
572                            Lillydale Lake parkrun
633                               Maribyrnong parkrun
651                                  Pakenham parkrun
666                             Toolern Creek parkrun
682                               Frog Hollow parkrun
736                                 Parkville parkrun
751                                  Brimbank parkrun
766                                    Coburg parkrun
804                                     Jells parkrun
841                              Altona Beach parkrun
948                              Wyndham Vale parkrun
989                         

In [29]:
melb_parkruns.to_csv("melb_parkruns.csv", index=False)

In [30]:
f"The number of parkruns in the Greater Melbourne area is {len(melb_parkruns)}."

'The number of parkruns in the Greater Melbourne area is 49.'

## Want to determine if I can call Melbourne Australia's parkrun capital

So I'm going to compare the number of parkruns in Greater Melbourne with the number in Great Sydney and Greater Brisbane respectively.

In [31]:
# Greater Sydney first.

greater_sydney_boundary = gpd.read_file("OSM_greater_sydney.geojson")
greater_sydney_boundary

geometry
0  MULTIPOLYGON (((150.27091 -33.67212, 150.27093...

In [32]:
greater_sydney_boundary.crs

<Geographic 2D CRS: EPSG:4326>
Name: WGS 84
Axis Info [ellipsoidal]:
- Lat[north]: Geodetic latitude (degree)
- Lon[east]: Geodetic longitude (degree)
Area of Use:
- name: World.
- bounds: (-180.0, -90.0, 180.0, 90.0)
Datum: World Geodetic System 1984 ensemble
- Ellipsoid: WGS 84
- Prime Meridian: Greenwich

In [33]:
syd_parkruns = gpd.sjoin(au_pk_gdf, greater_sydney_boundary, predicate='within')
syd_parkruns

eventname  \
21                     stpeters   
347                  parramatta   
366                    curlcurl   
502                      mosman   
591                  cooksriver   
600                campbelltown   
612                penrithlakes   
622                   rousehill   
637                      lawson   
718                    cronulla   
783                       menai   
806              chippingnorton   
840                    theponds   
842                      rhodes   
970                     panania   
1073                    galston   
1079                 willoughby   
1082                nepeanriver   
1084                      kamay   
1093                  rootyhill   
1150                 dollspoint   
1212               eastrichmond   
1260                   greenway   
1339                 wildflower   
1375                northsydney   
1483    cowpasturereservecamden   
1509                 centennial   
1731     werringtonlakesreserve   
1908            casulaparklands   
1980              whalanreserve   
2062            wentworthcommon   
2532  aubotanicgardenmountannan   
2546                    pirrama   
2584        macquarieuniversity   

                                      EventLongName  \
21                                St Peters parkrun   
347                              Parramatta parkrun   
366                               Curl Curl parkrun   
502                                  Mosman parkrun   
591                             Cooks River parkrun   
600                            Campbelltown parkrun   
612                           Penrith Lakes parkrun   
622                              Rouse Hill parkrun   
637                                  Lawson parkrun   
718                                Cronulla parkrun   
783                                   Menai parkrun   
806                         Chipping Norton parkrun   
840                               The Ponds parkrun   
842                                  Rhodes parkrun   
970                                 Panania parkrun   
1073                                Galston parkrun   
1079                             Willoughby parkrun   
1082                           Nepean River parkrun   
1084                                  Kamay parkrun   
1093                             Rooty Hill parkrun   
1150                            Dolls Point parkrun   
1212                          East Richmond parkrun   
1260                               GreenWay parkrun   
1339                             Wildflower parkrun   
1375                           North Sydney parkrun   
1483             Cowpasture Reserve parkrun, Camden   
1509                             Centennial parkrun   
1731               Werrington Lakes Reserve parkrun   
1908                       Casula Parklands parkrun   
1980                         Whalan Reserve parkrun   
2062                       Wentworth Common parkrun   
2532  Australian Botanic Garden Mount Annan parkrun   
2546                                Pirrama parkrun   
2584                   Macquarie University parkrun   

                             EventShortName LocalisedEventLongName  \
21                                St Peters                   None   
347                              Parramatta                   None   
366                               Curl Curl                   None   
502                                  Mosman                   None   
591                             Cooks River                   None   
600                            Campbelltown                   None   
612                           Penrith Lakes                   None   
622                              Rouse Hill                   None   
637                                  Lawson                   None   
718                                Cronulla                   None   
783                                   Menai                   None   
806                         Ch

In [34]:
len(syd_parkruns)

34

In [35]:
# Then Greater Brisbane.

greater_brisbane_boundary = gpd.read_file("OSM_greater_brisbane.geojson")
greater_brisbane_boundary

geometry
0  MULTIPOLYGON (((152.67969 -27.37226, 152.68418...

In [36]:
greater_brisbane_boundary.crs

<Geographic 2D CRS: EPSG:4326>
Name: WGS 84
Axis Info [ellipsoidal]:
- Lat[north]: Geodetic latitude (degree)
- Lon[east]: Geodetic longitude (degree)
Area of Use:
- name: World.
- bounds: (-180.0, -90.0, 180.0, 90.0)
Datum: World Geodetic System 1984 ensemble
- Ellipsoid: WGS 84
- Prime Meridian: Greenwich

In [37]:
bris_parkruns = gpd.sjoin(au_pk_gdf, greater_brisbane_boundary, predicate='within')
bris_parkruns

eventname                  EventLongName   EventShortName  \
177          newfarm               New Farm parkrun         New Farm   
365           wynnum                 Wynnum parkrun           Wynnum   
380         sandgate               Sandgate parkrun         Sandgate   
381        southbank             South Bank parkrun       South Bank   
434       mitchelton             Mitchelton parkrun       Mitchelton   
482   rocksriverside        Rocks Riverside parkrun  Rocks Riverside   
557         minnippi               Minnippi parkrun         Minnippi   
567        calamvale              Calamvale parkrun        Calamvale   
623     stonescorner          Stones Corner parkrun    Stones Corner   
625          wishart                Wishart parkrun          Wishart   
694          stlucia               St Lucia parkrun         St Lucia   
701        chermside              Chermside parkrun        Chermside   
827       forestlake            Forest Lake parkrun      Forest Lake   
904         ashgrove               Ashgrove parkrun         Ashgrove   
994           kedron                 Kedron parkrun           Kedron   
1147    mansfieldqld  Mansfield parkrun, Queensland    Mansfield QLD   
1211            lota                   Lota parkrun             Lota   
1271        zillmere               Zillmere parkrun         Zillmere   
1297     kelvingrove           Kelvin Grove parkrun     Kelvin Grove   
1416         pallara                Pallara parkrun          Pallara   

     LocalisedEventLongName  countrycode  seriesid  \
177                    None            3         1   
365                    None            3         1   
380                    None            3         1   
381                    None            3         1   
434                    None            3         1   
482                    None            3         1   
557                    None            3         1   
567                    None            3         1   
623                    None            3         1   
625                    None            3         1   
694                    None            3         1   
701                    None            3         1   
827                    None            3         1   
904                    None            3         1   
994                    None            3         1   
1147                   None            3         1   
1211                   None            3         1   
1271                   None            3         1   
1297                   None            3         1   
1416                   None            3         1   

                              EventLocation                     geometry  \
177                      New Farm, Brisbane  POINT (153.05164 -27.47113)   
365             Wynnum Manly Foreshore, QLD   POINT (153.1779 -27.44293)   
380                       Arthur Davis Park  POINT (153.06823 -27.31129)   
381                    South Bank Parklands  POINT (153.02396 -27.47887)   
434                            Teralba Park   POINT (152.9793 -27.40442)   
482                    Rocks Riverside Park  POINT (152.95924 -27.54263)   
557                       Meadowfields Road  POINT (153.11868 -27.49331)   
567                 Calamvale District Park  POINT (153.03929 -27.62078)   
623                             Hanlon Park  POINT (153.04247 -27.49872)   
625                  Wishart Community Park  POINT (153.10347 -27.56098)   
694                University of Queensland   POINT (153.0178 -27.49512)   
701   Kidspace Playground, 7th Brigade Park  POINT (153.03351 -27.37896)   
827               Forest Lake Sports Fields  POINT (152.96774 -27.61964)   
904                   Ashgrove Sportsground  POINT (152.97298 -27.44233)   
994                                  Kedron   POINT (153.0463 -27.40693)   
1147                           Tillack Park  POINT (153.10174 -27.52299)   
1211                   Lota Camping Reserve  POINT (153.19322 -27.46856

In [38]:
len(bris_parkruns)

20